lib/Dialect/Secret/Transforms/DistributeGeneric.cpp

```diff
@@ -569,15 +570,19 @@ struct SplitGeneric : public OpRewritePattern<GenericOp> {
       LLVM_DEBUG(opToDistribute->emitRemark()
                  << "Distributing through region holding op isolated in its "
                     "own generic\n");
-      return distributeThroughRegionHoldingOp(op, *opToDistribute, rewriter);
-    }
-
-    if (first) {
+      LogicalResult result =
+          distributeThroughRegionHoldingOp(op, *opToDistribute, rewriter);
+      if (failed(result)) {
+        return failure();
+      }
+    } else if (first) {
       splitGenericAfterFirstOp(op, rewriter);
     } else {
       splitGenericBeforeOp(op, *opToDistribute, rewriter);
     }
-    return success();
+
+    // Rerun secretness analysis
+    return solver->initializeAndRun(top);
```

Comment:
noticing extra timeouts here - maybe i should only run it when we know the loop vars have changed...

Reply:
In order to correctly re-run the analysis, a `solver->eraseAllStates();` is needed, unless we are superclear previous states does not affect the result of the current run. For secretness anlaysis it might be the case, but I'n not so sure so add a note here.

Reply:
So I think I should move the new lattice analysis only to the new and modified generics that have been created in the matchAndRewrite - the previous state shouldn't be changed since we don't change anything outside of the generic we match on.

That makes me wonder whether we just run initializeAndRun starting from the modified generics - since the timeout is happening from re-running on the moduleop which can be really large. This will only work if the secretness analysis properly initializes the entry states of the generic's secret args, which it does seem to.

Reply:
> That makes me wonder whether we just run initializeAndRun starting from the modified generics

This approach should work. I agree that "secretness analysis properly initializes the entry states of the generic's secret args", given the current `setToEntryState` of secretness analysis infers secretness from type and parenting op.